analyzers/src/SonarAnalyzer.CSharp/Rules/AssertionArgsShouldBePassedInCorrectOrder.cs

```diff
@@ -18,63 +18,87 @@
+                            : Enumerable.Empty<WrongArguments?>())
+                    .FirstOrDefault(x => x is not null) is (Expected: var expected, Actual: var actual))
             {
-                var methodCall = (InvocationExpressionSyntax)c.Node;
-                if (!methodCall.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression)
-                    || methodCall.ArgumentList.Arguments.Count < 2)
-                {
-                    return;
-                }
+                c.ReportIssue(Diagnostic.Create(Rule, CreateLocation(expected, actual)));
+            }
+        },
+        SyntaxKind.InvocationExpression);
 
-                var firstArgument = methodCall.ArgumentList.Arguments[0];
-                var secondArgument = methodCall.ArgumentList.Arguments[1];
-                if (firstArgument.Expression is LiteralExpressionSyntax
-                    || secondArgument.Expression is not LiteralExpressionSyntax)
+    private static KnownAssertParameters[] GetParameters(string name) =>
+        name switch
+        {
+            "AreEqual" => new KnownAssertParameters[]
                 {
-                    return;
-                }
-
-                var methodCallExpression = (MemberAccessExpressionSyntax)methodCall.Expression;
-
-                var methodKnownTypes = MethodsWithType.GetValueOrDefault(methodCallExpression.Name.Identifier.ValueText);
-                if (methodKnownTypes == null)
+                    new(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, "expected", "actual"),
+                    new(KnownType.NUnit_Framework_Assert, "expected", "actual")
+                },
+            "AreNotEqual" => new KnownAssertParameters[]
                 {
-                    return;
-                }
-
-                var symbolInfo = c.SemanticModel.GetSymbolInfo(methodCallExpression.Expression).Symbol;
-                var isAnyTrackedAssertType = (symbolInfo as INamedTypeSymbol).IsAny(methodKnownTypes);
-                if (!isAnyTrackedAssertType)
+                    new(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, "notExpected", "actual"),
+                    new(KnownType.NUnit_Framework_Assert, "expected", "actual")
+                },
+            "AreSame" => new KnownAssertParameters[]
                 {
-                    return;
-                }
+                    new(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, "expected", "actual"),
+                    new(KnownType.NUnit_Framework_Assert, "expected", "actual")
+                },
+            "AreNotSame" => new KnownAssertParameters[]
+                {
+                    new(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert, "notExpected", "actual"),
+                    new(KnownType.NUnit_Framework_Assert, "expected", "actual")
+                },
+            "Equal" or "Same" or "NotSame" => new KnownAssertParameters[]
```

Comment:
Is there any reason why [NotEqual](https://github.com/xunit/assert.xunit/blob/35047796d98e1250ef3d280c5ba6c4758c76a358/EqualityAsserts.cs#L347) is missing in this list?